contracts/token/BaseTokenSweep.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+/**
+ * @title
+ *
+ * @dev
+ */
+abstract contract BaseTokenSweep is UUPSUpgradeable {
+    address public beneficiary;
+
+    event BeneficiaryUpdated(address indexed beneficiary);
+
+    /**
+     * @notice Set the beneficiary of the token sweep
+     *
+     * @dev Needs access control implemented in the inheriting contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    function setBeneficiary(address _beneficiary) external virtual {
+        require(_beneficiary != address(0), "WithdrawTokens/beneficiary-zero");
```

Comment:
nit; lets keep the error message format consistent i.e.  "contract:description"

The existing style that I've been using is aligned with OZ messages. 
e.g.  "Mediator: creator not a contract"

"WithdrawTokens/beneficiary-zero" -> "BaseTokenSweep: beneficiary zero"

wdyt?

Reply:
Sounds good!